paddle/fluid/framework/ir/xpu/pad2d_xpu_fuse_pass.cc

```diff
@@ -0,0 +1,269 @@
+    GET_IR_NODE_FROM_SUBGRAPH(squeeze, squeeze, pattern);
+
+    auto* scope = param_scope();
+    PADDLE_ENFORCE_NOT_NULL(
+        scope, platform::errors::InvalidArgument("Scope cannot be nullptr."));
+
+    // pad2d
+    auto* block = pad3d->Op()->Block();
+    framework::OpDesc pad2d_op_desc(block);
+    pad2d_op_desc.SetType("pad2d_xpu");
+
+    // get pad3d's data_format
+    std::string data_format =
+        pad3d->Op()->GetAttrIfExists<std::string>("data_format");
+    // Judge unsqueeze && squeeze,  if axes is same, then eliminate
+    std::vector<int> unsqueeze_axes =
+        PADDLE_GET_CONST(std::vector<int>, unsqueeze->Op()->GetAttr("axes"));
+    std::vector<int> squeeze_axes =
+        PADDLE_GET_CONST(std::vector<int>, squeeze->Op()->GetAttr("axes"));
+    bool elimination = (unsqueeze_axes.size() == 1 &&
+                        ((data_format == "NCDHW" && unsqueeze_axes[0] == 2 &&
+                          squeeze_axes[0] == 2) ||
+                         (data_format == "NDHWC" && unsqueeze_axes[0] == 1 &&
+                          squeeze_axes[0] == 1)));
+    if (!elimination) return;
+
+    // set pad3d's data_format to pad2d's data_format
+    if (data_format == "NCDHW") {
+      data_format = "NCHW";
+    } else if (data_format == "NDHWC") {
+      data_format = "NHWC";
+    }
+
+    pad2d_op_desc.SetInput("x", {unsqueeze_out->Var()->Name()});
+    pad2d_op_desc.SetAttr("data_format", data_format);
+    pad2d_op_desc.SetAttr("mode", pad3d->Op()->GetAttr("mode"));
+    pad2d_op_desc.SetAttr("pad_value", pad3d->Op()->GetAttr("value"));
+    pad2d_op_desc.SetAttr(
+        "paddings", pad3d->Op()->GetAttrIfExists<std::vector<int>>("paddings"));
+
+    VarDesc pad2d_out_desc(pad3d_out->Var()->Name());
+    auto pad2d_paddings =
+        pad2d_op_desc.GetAttrIfExists<std::vector<int>>("paddings");
+    int lr = pad2d_paddings[0] + pad2d_paddings[1];  // left+right
+    int tb = pad2d_paddings[2] + pad2d_paddings[3];  // top+bottom
+    auto pad2d_shape = unsqueeze_input->Var()->GetShape();
+    if (data_format == "NCHW") {
+      pad2d_shape[2] += tb;  // height
+      pad2d_shape[3] += lr;  // width
+    } else {
+      // NHWC
+      pad2d_shape[1] += tb;  // height
+      pad2d_shape[2] += lr;  // width
+    }
+    pad2d_out_desc.SetShape(pad2d_shape);
+    pad2d_out_desc.SetDataType(proto::VarType::Type::VarType_Type_FP32);
+    auto pad2d_out = graph->CreateVarNode(&pad2d_out_desc);
+
+    pad2d_op_desc.RenameInput(unsqueeze_out->Var()->Name(),
+                              unsqueeze_input->Var()->Name());
+    pad2d_op_desc.SetOutput("out", {pad2d_out->Var()->Name()});
+
+    auto* pad2d_op = graph->CreateOpNode(&pad2d_op_desc);
+    pad2d_op_desc.Flush();
+    IR_NODE_UNLINK(unsqueeze_input, pad3d);
+    IR_NODE_UNLINK(pad3d, pad3d_out);
+    IR_NODE_UNLINK(pad3d_out, squeeze_out);
+    IR_NODE_LINK_TO(unsqueeze_input, pad2d_op);
+    IR_NODE_LINK_TO(pad2d_out, pad2d_op);
```

Comment:
```suggestion
    IR_NODE_UNLINK(unsqueeze_input, pad3d);
    IR_NODE_UNLINK(pad3d, pad3d_out);
    IR_NODE_UNLINK(pad3d_out, squeeze_out);
    IR_NODE_LINK_TO(unsqueeze_input, pad2d_op);
    IR_NODE_LINK_TO(pad2d_out, pad2d_op);
```
IR_NODE_LINK_TO(unsqueeze_input, pad2d_op);
IR_NODE_LINK_TO(pad2d_op, squeeze_out);